src/native/aws_signing.c

```diff
@@ -185,6 +188,10 @@ static void s_aws_signing_complete(struct aws_signing_result *result, int error_
 
     (*env)->DeleteLocalRef(env, java_signed_request);
 
+    /* I have no idea what we should do here... but the JVM really doesn't like us NOT calling this function after
+       we cross the barrier. */
+    (*env)->ExceptionCheck(env);
```

Comment:
Almost everywhere else we do this it's wrapped in AWS_FATAL_ASSERT()
should probably do the same here for consistency if nothing else